editoast/src/models/timetable.rs

```diff
@@ -125,6 +125,14 @@ impl Timetable {
                     base: result_train.mechanical_energy_consumed,
                     eco,
                 };
+                let arrival_time = result_train_eco.as_ref().map_or(arrival_time, |eco| {
+                    eco.0
+                        .head_positions
+                        .last()
+                        .expect("Train should have at least one position")
+                        .time
+                        + train_schedule.departure_time
+                });
```

Comment:
We can combine the two variable assignments into one.
There is no reason to define a first value on line 115 and replace it on line 128.

```suggestion
                let arrival_time = if let Some(eco) = result_train_eco {
                    eco.0.head_positions.last()
                } else {
                    result_train.head_positions.last()
                }
                .expect("Train should have at least one position")
                .time
                    + train_schedule.departure_time;
```